v3/src/utilities/data-utils.ts

```diff
@@ -42,3 +43,67 @@ export const numericSortComparator = function ({a, b, order}: ICompareProps): nu
   if (aIsNaN) return order === "asc" ? -1 : 1
   return order === "asc" ? a - b : b - a
 }
+
+export const
+  kTypeError = 1,
+  kTypeNaN = 2,
+  kTypeNull = 3,
+  kTypeString = 4,
+  kTypeBoolean = 5,
+  kTypeNumber = 6,
+  kTypeDate = 7,
+  // kTypeSimpleMap = 8, // e.g. boundaries
+  kTypeUnknown = 9
+
+export function typeCode(value: any) {
+  if (value == null) return kTypeNull
+  if (value instanceof Error) return kTypeError
+  if (isDate(value) || isDateString(value)) return kTypeDate
+  // if (value instanceof DG.SimpleMap) return kTypeSimpleMap;
+  switch (typeof value) {
+    case 'number': return isNaN(value) ? kTypeNaN : kTypeNumber
+    case 'boolean': return kTypeBoolean
+    case 'string': return kTypeString
+    /* istanbul ignore next */
+    default: return kTypeUnknown
+  }
+}
+
+export function sortableValue(value: any) {
+  const type = typeCode(value)
+  let num = type === kTypeNumber ? value : NaN
+  // strings convertible to numbers are treated numerically
+  if (type === kTypeString && value.length) {
```

Comment:
Shouldn't this check if the attribute has a manually set type? For example if the attribute type is string then I'd expect sorting `["001","001a"]` to treat the `"001"` as a string. If I'm reading the logic right here that array would end up reversed without taking into account the attribute type.